File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i,j = 0,len(numbers) - 1
        while(i != j):
            tsum = numbers[i] + numbers[j]

            if tsum > target:
                j -= 1
                continue
            
            if tsum < target:
                i += 1
                continue
            
            return [i+1,j+1]
        return [i+1,j+1]
            
        
            

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        return searchRec(nums,target,l,r);
    }

    int searchRec(vector<int>& nums, int target, int l, int r) {
        if(l > r)return - 1;
        int mid = (l + r)/2;
        if(nums[mid] == target)
            return mid;
        if(nums[mid] < target)
            return searchRec(nums, target, mid + 1, r);
        else
            return searchRec(nums, target, l, mid - 1);
    }

};
